addons/portal/controllers/mail.py

```diff
@@ -112,6 +112,19 @@ def _portal_post_has_content(self, res_model, res_id, message, attachment_ids=No
         """ Tells if we can effectively post on the model based on content. """
         return bool(message) or bool(attachment_ids)
 
+    @http.route('/mail/avatar/mail.message/<int:res_id>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
+    def portal_avatar(self, res_id=None, height=50, width=50, access_token=None):
+        """ Get the avatar image in the chatter of the portal """
+        record_sudo = request.env.ref('web.image_placeholder').sudo()
+        if access_token and res_id:
+            mail_record = request.env['mail.message'].browse(res_id).sudo()
+            access_as_sudo = _check_special_access(mail_record.model, mail_record.res_id, token=access_token)
+            if access_as_sudo and mail_record:
+                record_sudo = mail_record
+
+        stream = request.env['ir.binary'].sudo()._get_image_stream_from(record_sudo, field_name='author_avatar', width=int(width), height=int(height))
```

Comment:
This `sudo` is not needed
```suggestion
        stream = request.env['ir.binary']._get_image_stream_from(record_sudo, field_name='author_avatar', width=int(width), height=int(height))
```